src/adb.js

```diff
@@ -1 +1,393 @@
+	 */
+	forward(deviceId, src, dest) {
+		if (typeof deviceId !== 'string' || !deviceId) {
+			return Promise.reject(new TypeError('Expected device ID to be a string.'));
+		}
+
+		src = util.expandPath(src);
+		if (!util.existsSync(src)) {
+			return Promise.reject(new TypeError(`Source file "${src}" does not exist.`));
+		}
+
+		return sdk
+			.detect(this.opts)
+			.then(result => util.run(result.sdk.executables.adb, ['-s', deviceId, 'forward', src, dest]));
+	}
+
+	/**
+	 * Pushes a single file to a device or emulator.
+	 *
+	 * @param {String} deviceId - The id of the device or emulator.
+	 * @param {String} src - The source file to copy to the device.
+	 * @param {String} dest - The destination to write the file.
+	 * @returns {Promise}
+	 * @access public
+	 */
+	push(deviceId, src, dest) {
+		if (typeof deviceId !== 'string' || !deviceId) {
+			return Promise.reject(new TypeError('Expected device ID to be a string.'));
+		}
+
+		src = util.expandPath(src);
+		if (!util.existsSync(src)) {
+			return Promise.reject(new TypeError(`Source file "${src}" does not exist.`));
+		}
+
+		return sdk
+			.detect(this.opts)
+			.then(result => util.run(result.sdk.executables.adb, ['-s', deviceId, 'push', src, dest]));
+	}
+
+	/**
+	 * Pulls a single file from a device or emulator.
+	 *
+	 * @param {String} deviceId - The id of the device or emulator.
+	 * @param {String} src - The source file to copy from the device.
+	 * @param {String} dest - The destination to write the file.
+	 * @returns {Promise}
+	 * @access public
+	 */
+	pull(deviceId, src, dest) {
+		if (typeof deviceId !== 'string' || !deviceId) {
+			return Promise.reject(new TypeError('Expected device ID to be a string.'));
+		}
+
+		if (typeof dest !== 'string' || !dest) {
+			return Promise.reject(new TypeError('Expected destination to be a string.'));
+		}
+
+		src = util.expandPath(src);
+		if (!util.existsSync(src)) {
+			return Promise.reject(new TypeError(`Source file "${src}" does not exist.`));
+		}
+
+		dest = util.expandPath(dest);
+		const destDir = path.dirname(dest);
+
+		mkdirp.sync(destDir);
+
+		return sdk
+			.detect(this.opts)
+			.then(result => util.run(result.sdk.executables.adb, ['-s', deviceId, 'pull', src, dest]));
+	}
+
+	/**
+	 * Returns the pid of the specified app for the specified device/emulator, if running.
+	 *
+	 * @param {String} deviceId - The id of the device or emulator.
+	 * @param {String} appId - The application's id.
+	 * @returns {Promise}
+	 * @access public
+	 */
+	getPid(deviceId, appId) {
+		if (typeof deviceId !== 'string' || !deviceId) {
+			return Promise.reject(new TypeError('Expected device ID to be a string.'));
+		}
+
+		if (typeof appId !== 'string' || !appId) {
+			return Promise.reject(new TypeError('Expected app ID to be a string.'));
+		}
+
+		return this
+			.shell(deviceId, 'ps')
+			.then(data => {
+				if (data) {
+					const lines = data.toString().split('\n');
+					for (let i = 0, j = lines.length; i < j; i++ ) {
```

Comment:
Instead of a for loop, use a for-of loop!

    for (let line of lines) {
        // Use `lines` instead of `lines[i]`
    }